XTS/WS-T/dev/src/org/jboss/jbossts/xts/recovery/participant/at/XTSATRecoveryManager.java

```diff
@@ -23,12 +25,25 @@ public abstract class XTSATRecoveryManager {
      */
     public static XTSATRecoveryManager getRecoveryManager()
     {
-        int i = 0;
-
-        while (theRecoveryManager == null && i++ < 2) {
-            try {
-                Thread.sleep(1000);
-            } catch (InterruptedException e) {
+        long start = System.currentTimeMillis();
+        long end = start + MAX_INIT_TIMEOUT_MILLIS;
+
+        synchronized (lock) {
+            while (theRecoveryManager == null) {
+                try {
+                    lock.wait(MAX_INIT_TIMEOUT_MILLIS);
+                } catch (InterruptedException e) {
+                    // reset the interrupted status
+                    Thread.currentThread().interrupt();
+                }
+
+                if (System.currentTimeMillis() > end) {
+                    if (theRecoveryManager == null) {
+                        WSTLogger.logger.warnf("%s recoveryManager has still not been set after %d ms",
+                                XTSBARecoveryManager.class, MAX_INIT_TIMEOUT_MILLIS);
```

Comment:
@mmusgrov I think this should be XTSATRecoveryManager and the import on line 5 removed, do you agree?

Reply:
My IDE needs intelligent cut and paste ;-)